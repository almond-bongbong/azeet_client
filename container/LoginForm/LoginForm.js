import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styled from 'styled-components';
import kakaoLoginImage from 'static/images/kakao_login_btn.png';
import { alignXY } from 'style/mixin';
import { useKakao } from 'hooks';
import { toastActions } from 'store/modules/toast';
import { pendingActions } from 'store/modules/pending';
import { authActions, authTypes } from 'store/modules/auth';
import SimpleLoader from 'components/Loader/SimpleLoader';

const LoginFormStyle = styled.div`
  ${alignXY};
  text-align: center;
  & p {
    margin-bottom: 15px;
  }
  & button {
    display: block;
  }
  & .buttons img {
    height: 49px;
  }
  & .btn_kakao {
    overflow: hidden;
    width: 226px;
    height: 48px; 
    border-radius: 4px;
    background-color: #f7e317;
  }
`;

const LoginForm = () => {
  const Kakao = useKakao();
  const dispatch = useDispatch();
  const kakoPending = useSelector(state => state.pending[authTypes.LOGIN_WITH_KAKAO]);
  const user = useSelector(state => state.auth.user);

  // if (process.browser && user) Router.replace('/');

  const loginKakao = (authResponse) => {
    dispatch(authActions.loginWithKakao(authResponse.access_token));
  };

  const handleKakao = () => {
    dispatch(pendingActions.pending(authTypes.LOGIN_WITH_KAKAO));
    Kakao.Auth.login({
      success: loginKakao,
      fail: () => {
        dispatch(toastActions.toast('문제가 발생했습니다'));
        dispatch(pendingActions.finally(authTypes.LOGIN_WITH_KAKAO));
      },
    });
  };

  return (
    <LoginFormStyle>
      <p>SNS 계정으로 간편하게 시작해 보세요</p>
      <div className="buttons">
        <button className="btn_kakao" type="button" title="kakao login" onClick={handleKakao}>
          {kakoPending ? (
            <SimpleLoader size={25} />
          ) : (
            <img src={kakaoLoginImage} alt="Log in with Kakao" />
          )}
        </button>
      </div>
    </LoginFormStyle>
  );
};

export default LoginForm;
