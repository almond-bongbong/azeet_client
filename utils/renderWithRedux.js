import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configStore from 'store/configure';
import Alert from 'components/Alert';
import Confirm from 'components/Confirm';
import Toast from 'components/Toast';

function renderWithRedux(ui, initialState) {
  const store = configStore(initialState);
  const utils = render(
    <Provider store={store}>
      {ui}
      <Alert />
      <Confirm />
      <Toast />
    </Provider>,
  );
  return {
    ...utils,
    store,
  };
}

export default renderWithRedux;
