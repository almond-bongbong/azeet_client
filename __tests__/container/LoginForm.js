import React from 'react';
import renderWithRedux from 'utils/renderWithRedux';
import LoginForm from 'container/LoginForm';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

describe('LoginForm', () => {
  const mock = new MockAdapter(axios, { delayResponse: 200 }); // 200ms 가짜 딜레이 설정
  mock.onPost('localhost:8080/auth/kakao').reply(200, {
    accessToken: 'abc',
    tokenType: 'Bearer',
  });

  it('matches snapshot', () => {
    const utils = renderWithRedux(<LoginForm />);
    expect(utils.container).toMatchSnapshot();
  });

  it('login form', () => {
    const { getByTitle, getByAltText } = renderWithRedux(<LoginForm />);
    getByTitle('kakao login');
    getByAltText('Log in with Kakao');
  });
});
