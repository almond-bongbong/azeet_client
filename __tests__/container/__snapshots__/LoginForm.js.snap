// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm matches snapshot 1`] = `
<div>
  <div
    class="LoginForm__LoginFormStyle-sc-13qd6qc-0 jNUouh"
  >
    <p>
      SNS 계정으로 간편하게 시작해 보세요
    </p>
    <div
      class="buttons"
    >
      <button
        class="btn_kakao"
        title="kakao login"
        type="button"
      >
        <img
          alt="Log in with Kakao"
          src="[object Object]"
        />
      </button>
    </div>
  </div>
</div>
`;
